### DISCUSSION TITLE
How to construct `SyntaxContext` from usize

### DISCUSSION THREAD

**Discussion by hyf0** [swc-project/swc#3285] [Q&A]
I saw these codes were commented out.
 https://github.com/swc-project/swc/blob/main/crates/swc_common/src/syntax_pos/hygiene.rs#L203-L209

What's the relationship between `SyntaxContext` and `Mark`?. It looks like `SyntaxContext` must be derived from a `Mark`.

Or Is there any material I could learn about it?

**Comment by kdy1** [comment]
`SyntaxContext` contains 0+ marks. `SyntaxContext::empty()` has 0 mark, and you can add `Mark` by calling `.apply(mark)`.

You should not create `SyntaxContext` from usize.
